### PR TITLE
Save ip address + User service auth

### LINKED ISSUE #61: Implement UserLoginService
* v soucasne chvili je to pouze scaffold, ktery da pristup kazdemu, je treba implementovat jak user model v databazi tak prave pristup a verifikaci
* s tim se poji i registrace uzivatele
* a take spravne hashovani hesel

### LINKED ISSUE #74: Save IP address of request
Do DB si chceme ukládat i IP adresu, ze které nám data přišla. Více méně jenom pro evidenční účely a řešení případných problémů se zneužitím identity. Možná také na rozlišení, že se člověk hlásí z Prahy 7

### LINKED ISSUE #79: Performance testing
* zjistit jakej mame vykon (number of req/minute) pri soucasne implementaci na soucasnem (2 cores + 4GB) AWS stroji
* jednoduchy python/kotlin script, ktery zavola cele flow - login + get insurance + register patient
* prerekvizita je zavreny #78 a nasazeny system na AWS

### PR DESCRIPTION
**Pull request by LukasForst**
* closes #74 
* closes #61 
* closes #79

### CODE REVIEW

File: backend/src/main/kotlin/blue/mild/covid/vaxx/service/UserService.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        } else {
+            throw CredentialsMismatchException()
+        }
+    }
+
+    suspend fun registerUser(registration: UserRegistrationDtoIn): Unit = newSuspendedTransaction {
+        User.insert {
+            it[id] = UUID.randomUUID().toString()
+            it[username] = registration.username
+            it[passwordHash] = passwordHashProvider.hashPassword(registration.password)
```
Comment (marazt) [inline backend/src/main/kotlin/blue/mild/covid/vaxx/service/UserService.kt:37]: Pridal bych sem validaci hesla, tj. aby ho zadaval 2x a my to overovali. Minimalne bych to delal na FE, kdyz ne na BE.

File: backend/src/main/resources/db/migration/V1__create_first_scheme.sql [hunk 2]
```diff
@@ -102,3 +103,27 @@ CREATE TRIGGER tgr_answers_set_updated
     ON answers
     FOR EACH ROW
 EXECUTE PROCEDURE set_updated();
+
+-- Create table users
+CREATE TABLE users
+(
+    id            VARCHAR(36) UNIQUE  NOT NULL,
+    created       timestamptz         NOT NULL,
+    updated       timestamptz,
```
Comment (marazt) [inline backend/src/main/resources/db/migration/V1__create_first_scheme.sql:112]: Pridej sem teda do vsech updated NOT NULL, at je to konzistentni s Kotlin kodem.

File: backend/src/main/kotlin/blue/mild/covid/vaxx/service/UserService.kt (hunk 1 above)
Comment (LukasForst) [reply backend/src/main/kotlin/blue/mild/covid/vaxx/service/UserService.kt:37]: udelame bud na FE, nebo az budeme resit user management pres UI, scriptu je to jedno

File: backend/src/test/kotlin/blue/mild/covid/vaxx/platform/PerformanceTest.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+private val logger = createLogger("PerformanceTest")
+
+fun main() {
+//    val test = timeTest()
+    val test = requestsCountTest()
+    runBlocking { test.execute() }.also(::logResults)
+}
+
+private fun timeTest() = TimeLoadTest(
+    targetHost = "http://localhost:8080",
```
Comment (marazt) [inline backend/src/test/kotlin/blue/mild/covid/vaxx/platform/PerformanceTest.kt:16]: Move to constant please, maybe to ENV variable.

File: backend/src/test/kotlin/blue/mild/covid/vaxx/platform/PerformanceTest.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        username = "mildblue",
+        password = "bluemild"
+    ),
+    requestTimeoutsSeconds = 60,
+    coroutineWorkers = 1,
+    runningTimeInSeconds = 60
+)
+
+private fun requestsCountTest() = RoundsLoadTest(
+    targetHost = "http://localhost:8080",
```
Comment (marazt) [inline backend/src/test/kotlin/blue/mild/covid/vaxx/platform/PerformanceTest.kt:27]: Move to constant please, maybe to ENV variable.

File: backend/src/test/kotlin/blue/mild/covid/vaxx/platform/TimeLoadTest.kt [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    override suspend fun execute(): List<RequestMetric> {
+        val endTime = now.plusSeconds(runningTimeInSeconds)
+
+        coroutineScope {
+            repeat(coroutineWorkers) {
+                launch {
+                    while (isTimeLeft(endTime)) {
+                        runPatientRegistrationWithBuilder(
+                            { answers, insuranceCompany, confirmation ->
+                                PatientRegistrationDtoIn(
```
Comment (marazt) [inline backend/src/test/kotlin/blue/mild/covid/vaxx/platform/TimeLoadTest.kt:35]: Maybe create some func that returns this prepared test object.

**Review by marazt — DISMISSED** [review]

**Review by marazt — APPROVED** [review]

**Pull request merged**